addOns/ascanrulesAlpha/src/test/java/org/zaproxy/zap/extension/ascanrulesAlpha/Log4ShellScanRuleUnitTest.java

```diff
@@ -41,4 +55,74 @@ void shouldTargetJavaApps() {
         // Then
         assertThat(rule.targets(techSet), is(equalTo(true)));
     }
+
+    @Test
+    void shouldSendHttpMessageForEachPayload() throws Exception {
+        // Given
+        ExtensionOast oastExtension = setupTest();
+        when(oastExtension.registerAlertAndGetPayload(any())).thenReturn("PAYLOAD");
+
+        // When
+        rule.scan();
+
+        // Then
+        assertThat(httpMessagesSent, hasSize(Log4ShellScanRule.ATTACK_PATTERN_COUNT));
+    }
+
+    @Test
+    void whenIOExceptionOccurred_shouldSendHttpMessageForFurtherPayloads() throws Exception {
+        // Given
+        ExtensionOast oastExtension = setupTest();
+        when(oastExtension.registerAlertAndGetPayload(any()))
+                .thenReturn("PAYLOAD1")
+                .thenThrow(IOException.class)
+                .thenReturn("PAYLOAD");
+
+        // When
+        rule.scan();
+
+        // Then
+        assertThat(httpMessagesSent, hasSize(Log4ShellScanRule.ATTACK_PATTERN_COUNT - 1));
+    }
+
+    @Test
+    void whenArbitraryExceptionOccurred_shouldStopSendingFurtherPayloads() throws Exception {
+        // Given
+        ExtensionOast oastExtension = setupTest();
+        when(oastExtension.registerAlertAndGetPayload(any()))
+                .thenReturn("PAYLOAD1")
+                .thenThrow(NullPointerException.class)
+                .thenReturn("PAYLOAD");
+
+        // When
+        rule.scan();
+
+        // Then
+        assertThat(httpMessagesSent, hasSize(1));
+    }
+
+    private ExtensionOast setupTest() throws HttpMalformedHeaderException {
```

Comment:
Given that the majority of the tests will need this it could be moved to a before each and make `extensionOast` (be good to call it like the class name) part of the test state.